File: _deploy_onboarding_backend/execute-migration.js
const { Client } = require('pg');
const fs = require('fs');
const path = require('path');

const migrationPath = path.join(__dirname, 'migrations', '010_add_document_linking_support.sql');
const sql = fs.readFileSync(migrationPath, 'utf8');

// Supabase PostgreSQL connection
const client = new Client({
  host: 'db.hncvsextwmvjydcukdwx.supabase.co',
  port: 5432,
  database: 'postgres',
  user: 'postgres',
  password: 'Itbfr@p8ky#dRMAHLdi', // Using the master password
  ssl: { rejectUnauthorized: false }
});

async function runMigration() {
  try {
    console.log('🔄 Connecting to Supabase PostgreSQL...\n');
    await client.connect();
    console.log('✅ Connected successfully!\n');

    console.log('📋 Executing migration SQL...\n');
    const startTime = Date.now();
    
    await client.query(sql);
    
    const duration = Date.now() - startTime;
    
    console.log(`\n✅ Migration executed successfully in ${duration}ms!\n`);
    console.log('📊 Tables created/modified:');
    console.log('   ✅ inbox_attachments - extended with attachment_kind, document_type, linked_candidate_id');
    console.log('   ✅ candidate_documents - new table for linked documents');
    console.log('   ✅ unmatched_documents - new table for pending documents');
    console.log('   ✅ candidates - extended with document checklist fields');
    console.log('   ✅ Indexes created for performance');
    console.log('   ✅ Trigger function created for auto-updating checklist\n');
    
    console.log('🎉 Document auto-linking system is now fully operational!\n');

    // Verify tables were created
    console.log('🔍 Verifying tables...\n');
    
    const tables = await client.query(`
      SELECT table_name 
      FROM information_schema.tables 
      WHERE table_schema = 'public' 
      AND table_name IN ('candidate_documents', 'unmatched_documents')
      ORDER BY table_name;
    `);

    if (tables.rows.length === 2) {
      console.log('✅ All required tables exist:');
      tables.rows.forEach(row => {
        console.log(`   ✅ ${row.table_name}`);
      });
    }

    // Check for new columns
    const columns = await client.query(`
      SELECT column_name FROM information_schema.columns
      WHERE table_name = 'candidates' 
      AND column_name IN ('passport_received', 'cnic_received', 'degree_received', 'father_name')
      ORDER BY column_name;
    `);

    if (columns.rows.length > 0) {
      console.log('\n✅ Candidates table extended with:');
      columns.rows.forEach(row => {
        console.log(`   ✅ ${row.column_name}`);
      });
    }

    console.log('\n✨ Database is ready for document auto-linking!\n');

  } catch (err) {
    console.error('❌ Migration failed:', err.message);
    process.exit(1);
  } finally {
    await client.end();
  }
}

runMigration();

File: _deploy_onboarding_backend/scripts/archive/execute-migration.py
#!/usr/bin/env python3
import sys
import os

try:
    import psycopg2
except ImportError:
    print('Installing psycopg2...')
    import subprocess
    subprocess.run([sys.executable, '-m', 'pip', 'install', 'psycopg2-binary', '-q'])
    import psycopg2

# Load environment variables
from pathlib import Path
env_path = Path(__file__).parent.parent / '.env'
if env_path.exists():
    with open(env_path) as f:
        for line in f:
            if '=' in line and not line.startswith('#'):
                key, value = line.strip().split('=', 1)
                os.environ[key] = value

# Supabase connection
try:
    print('\n' + '='*70)
    print('🚀 DATABASE MIGRATION 011: ADD CV EXTRACTION FIELDS')
    print('='*70 + '\n')
    
    print('⏳ Connecting to Supabase PostgreSQL...')
    
    password = os.environ.get('SUPABASE_DB_PASSWORD', 'MonsterBurger123!!')
    
    conn = psycopg2.connect(
        host='db.hncvsextwmvjydcukdwx.supabase.co',
        port=5432,
        database='postgres',
        user='postgres',
        password=password,
        sslmode='require',
        connect_timeout=10
    )
    
    cursor = conn.cursor()
    print('✅ Connected!\n')
    
    # Read migration file
    with open('migrations/011_add_cv_extraction_fields.sql', 'r', encoding='utf-8') as f:
        sql = f.read()
    
    print('⏳ Executing migration SQL...\n')
    
    # Execute the SQL
    cursor.execute(sql)
    conn.commit()
    
    print('✅ Migration SQL executed successfully!\n')
    
    # Verify columns were created
    print('🔍 Verifying new columns...\n')
    
    verify_sql = """
        SELECT column_name 
        FROM information_schema.columns 
        WHERE table_name='candidates' 
        AND column_name IN (
            'nationality', 'position', 'experience_years', 'country_of_interest',
            'skills', 'languages', 'education', 'certifications',
            'previous_employment', 'passport_expiry', 'professional_summary',
            'extraction_confidence', 'extraction_source', 'extracted_at'
        )
        ORDER BY column_name
    """
    
    cursor.execute(verify_sql)
    columns = [row[0] for row in cursor.fetchall()]
    
    if columns:
        print(f'✅ SUCCESS! Found {len(columns)}/14 new columns:\n')
        for col in columns:
            print(f'   ✓ {col}')
        print('\n' + '='*70)
        print('✨ MIGRATION COMPLETED SUCCESSFULLY!')
        print('='*70 + '\n')
        
        # Check extraction_history table
        cursor.execute("SELECT COUNT(*) FROM information_schema.tables WHERE table_name='extraction_history'")
        if cursor.fetchone()[0] > 0:
            print('✅ extraction_history table created\n')
    else:
        print('⚠️  Warning: Could not verify new columns\n')
    
    cursor.close()
    conn.close()
    
    sys.exit(0)
    
except psycopg2.OperationalError as e:
    print(f'\n❌ Connection Error: {str(e)}\n')
    print('The secret key provided may not be the database password.')
    print('Please check your Supabase project settings for the correct password.\n')
    sys.exit(1)
    
except Exception as e:
    print(f'\n❌ Error: {str(e)}\n')
    import traceback
    traceback.print_exc()
    sys.exit(1)
